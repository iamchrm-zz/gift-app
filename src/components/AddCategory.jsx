import React, { useState } from "react";

function AddCategory({ setCategories }) {
  const [inputValue, setInputValue] = useState("");
  const handleInputChange = (e) => {
    console.log(e.target.value);
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    console.log("submit");

    const result =
      inputValue.trim().length > 2
        ? setCategories((valorAnterior) => [inputValue])
        : alert("Invalid");
    const reset = setInputValue("");

    return result + reset;
  };

  return (
    <form onSubmit={handleSubmit}>
      <div className="flex justify-center">
        <input
          type="text"
          className="py-4 px-4 pl-9 border border-slate-300 shadow-sm focus:outline-none focus:border-sky-500 focus:ring-sky-500 focus:ring-1  rounded-full text-lg"
          placeholder="Gifs Searcher"
          value={inputValue}
          onChange={handleInputChange}
        />
      </div>
    </form>
  );
}

export default AddCategory;
