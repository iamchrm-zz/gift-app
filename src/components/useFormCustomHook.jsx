import React from "react";

export const useFormCustomHook = () => {
  return (
    <form>
      <h1> Form with custom form hook</h1>

      <input type="text" name="name" placeholder="Set your name" onChange={}/>
        <hr />
        <input type="text" name="email" placeholder="your@email.com" onChange={}/>

    
    </form>
  );
};
