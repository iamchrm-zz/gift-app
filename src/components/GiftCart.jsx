import React, { useState } from "react";
import { getGifts } from "../helpers/getGifts";
import { SeletecItems } from "./GiftItem";

const GiftCart = () => {
  return (
    <div>
      <h1> hola</h1>
    </div>
  );
};

export default GiftCart;
