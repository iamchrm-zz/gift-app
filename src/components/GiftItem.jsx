import React, { useState } from "react";
import PropTypes from "prop-types";
import GiftCart from "./GiftCart";

export const GiftItem = ({ id, title, url }) => {
  return (
    <>
      <li key={id}>
        <div className="drop-shadow-xl">
          <img
            src={url}
            alt={title}
            className=" transition duration-400 ease-in-out hover:-translate-y-1 hover:scale-110 object-cover h-48 w-96 rounded-lg drop-shadow-xl"
          />
          <div>
            <button>add</button>
          </div>
        </div>
      </li>
    </>
  );
};

GiftItem.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
