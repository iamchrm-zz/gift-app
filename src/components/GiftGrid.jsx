import React from "react";

import { useFetchGifts } from "../hooks/useFetchGifts";
import { GiftItem } from "./GiftItem";

export const GiftGrid = ({ category, OnAdd }) => {
  const { data: images, loading } = useFetchGifts(category);

  return (
    <div>
      {loading ? (
        <div className="flex justify-center pt-10">
          <div className="transition ease-in duration-300 ">
            <div className="animate-pulse fadeIn">
              <h1 className="text-white text-center text-6xl">Loading</h1>
            </div>
          </div>
        </div>
      ) : (
        <div className="flex justify-center pt-4 ">
          <div className="transition ease-in duration-300 grid grid-cols-3 gap-4 pointer-events-auto">
            {images.map((img) => (
              <GiftItem key={img.id} {...img} OnAdd={OnAdd} />
            ))}
          </div>
        </div>
      )}
    </div>
  );
};
