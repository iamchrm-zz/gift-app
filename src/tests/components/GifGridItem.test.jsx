import React from "react";
import { shallow } from "enzyme";
import { GiftItem } from "../../components/GiftItem";

describe("Pruebas en GifGridItem", () => {
  const id = "1";
  const title = "title";
  const url = "http://";
  const clase = "drop-shadow-xl";
  const wrapper = shallow(<GiftItem id={id} title={title} url={url} />);
  test("Debe mostrar el component correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe tener una clase", () => {
    const div = wrapper.find("div");

    expect(div.prop("className")).toBe(clase);
  });
});
