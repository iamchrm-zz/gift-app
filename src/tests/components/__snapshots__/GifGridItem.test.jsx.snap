// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Debe mostrar el component correctamente 1`] = `
<Fragment>
  <li
    key="1"
  >
    <div
      className="drop-shadow-xl"
    >
      <img
        alt="title"
        className=" transition duration-400 ease-in-out hover:-translate-y-1 hover:scale-110 object-cover h-48 w-96 rounded-lg drop-shadow-xl"
        src="http://"
      />
    </div>
  </li>
</Fragment>
`;
